src/main.rs

```diff
@@ -44,15 +43,42 @@ fn main() {
 	}
 
 	let rootfs = matches.opt_str("r").unwrap_or_else(|| {
-		println!("Error: Please pass: --rootfs");
+		println!("Error: Please pass `--rootfs <path>`");
 		print_usage(&program, &opts);
 		process::exit(7);
 	});
-	let command_string = matches.opt_str("c").unwrap_or_else(|| {
-		println!("Error: Please pass: --command");
+
+	let c = matches.opt_str("c"); // NB: Seperate let binding for lifetime.
+	let (command, args) = determine_command_tuple(&c, &args[begin_unprocessable..args.len()]).ok().unwrap_or_else(|| {
+		println!("Error: Please pass `--command <shell command>` or `-- <command> <argument>...`");
 		print_usage(&program, &opts);
 		process::exit(7);
 	});
 
-	run(&rootfs, &command_string);
+	runtime::run_container(&rootfs, command, args);
+}
+
+/// Determines based on the inputs whether we are going to invoke a shell, with
+/// shell interpretation, or a simple unescaped argument vector.
+/// Rearranges arguments as needed, but doesn't reallocate.
+fn determine_command_tuple<'a, T: AsRef<str> + 'a>(shell_command: &'a Option<T>, argv: &'a [T]) -> Result<(&'a str, Vec<&'a str>), Box<dyn Error>> {
+	let mut vec: Vec<&str> = vec![];
+
+	// Prepend shell and command string if a command string is given.
+	if let Some(shell_command) = shell_command {
+		vec.push("/bin/sh");
```

Comment:
I think `/bin/sh` will take interactive commands as command string with -c flag. Otherwise this is going throw `/bin/sh: 0: Can't open <x>`. I think we can pass container commands as command string to /bin/sh with -C. I can open a quick PR in your branch.

Reply:
👍 

That's correct. Sorry to push an error like that. I had some fancier code previously for selecting the shell -- looking at `$SHELL` -- and removed it and must have made a mistake when reorganizing.